firewood/src/merkle/node/branch.rs

```diff
@@ -140,8 +142,8 @@ impl BranchNode {
     }
 
     pub(super) fn encode<S: ShaleStore<Node>>(&self, store: &S) -> Vec<u8> {
-        // TODO: add path to encoded node
-        let mut list = <[Encoded<Vec<u8>>; Self::MAX_CHILDREN + 1]>::default();
+        // path + children + value
```

Comment:
This comment should be re-ordered to match the order in which the values are encoded.